tools/ktx/validation_messages.h

```diff
@@ -0,0 +1,678 @@
+        7130, "Incompatible KTXanimData and KTXcubemapIncomplete metadata. KTXanimData and KTXcubemapIncomplete cannot be present at the same time.",
+        "Both KTXanimData and KTXcubemapIncomplete is present but they are incompatible."
+    };
+
+    static constexpr IssueError KTXastcDecodeModeMissingNull{
+        7131, "Invalid KTXastcDecodeMode metadata. The value is missing the NULL terminator.",
+        "The last byte of the value must be a NULL terminator."
+    };
+    static constexpr IssueError KTXastcDecodeModeInvalidValue{
+        7132, "Invalid KTXastcDecodeMode value. The value must be either \"rgb9e5\" or \"unorm8\".",
+        "The value is \"{}\" but the value must be either \"rgb9e5\" or \"unorm8\"."
+    };
+    static constexpr IssueError KTXastcDecodeModeunorm8NotLDR{
+        7133, "Invalid KTXastcDecodeMode value. \"unorm8\" is only valid for LDR formats.",
+        "The value is \"unorm8\" but it is invalid for non-LDR VkFormat {}."
+    };
+    static constexpr IssueWarning KTXastcDecodeModeNotASTC{
+        7134, "KTXastcDecodeMode has no effect on and should not be present in KTX files that use a non-ASTC formats.",
+        "KTXastcDecodeMode is present but for colorModel {} it has no effect."
+    };
+    static constexpr IssueWarning KTXastcDecodeModeWithsRGB{
+        7135, "KTXastcDecodeMode has no effect on and should not be present in KTX files that use the sRGB transfer function.",
+        "KTXastcDecodeMode is present but for transferFunction {} it has no effect."
+    };
+};
+
+// struct SGD {
+//     static constexpr IssueError UnexpectedSupercompressionGlobalData{
+//         80, "Supercompression global data found scheme that is not Basis."
+//     };
+//     static constexpr IssueError MissingSupercompressionGlobalData{
+//         80, "Basis supercompression global data missing."
+//     };
+//     static constexpr IssueError InvalidImageFlagBit{
+//         80, "Basis supercompression global data imageDesc.imageFlags has an invalid bit set."
+//     };
+//     static constexpr IssueError IncorrectGlobalDataSize{
+//         80, "Basis supercompression global data has incorrect size."
+//     };
+//     static constexpr IssueError ExtendedByteLengthNotZero{
+//         80, "extendedByteLength != 0 in Basis supercompression global data."
+//     };
+//     static constexpr IssueError DFDMismatchAlpha{
+//         80, "supercompressionGlobalData indicates no alpha but DFD indicates alpha channel."
+//     };
+//     static constexpr IssueError DFDMismatchNoAlpha{
+//         80, "supercompressionGlobalData indicates an alpha channel but DFD indicates no alpha channel."
+//     };
+// };
+//
+// struct System {
+//     static constexpr IssueError OutOfMemory{
+//         100, "System out of memory."
+//     };
+// };
+
+// -------------------------------------------------------------------------------------------------
+
+} // namespace ktx
```

Comment:
Messages are meant as static text while details are meant for dynamic text which include additional dynamic information. There are only a few cases where there is no additional dynamic information so I think we should always have both for consistency in the outputs.

However, its true that there are redundant parts. The initial concept was to always include every possible information in the static message. But instead of the cutting from the dynamic details we could rework and remove the rules/requirements from the static massages and only include those in the details.

Examples:
```C++
// Current
"sgdByteOffset must be 0 when sgdByteLength is 0.",
"sgdByteOffset is {} but must be 0 when sgdByteLength is 0."

// Proposed 1
"Invalid sgdByteOffset",
"sgdByteOffset is {} but must be 0 when sgdByteLength is 0."

// Proposed 2
"Invalid sgdByteOffset with sgdByteLength = 0",
"sgdByteOffset is {} but must be 0 when sgdByteLength is 0."

// Proposed 3
"Non-zero sgdByteOffset",
"sgdByteOffset is {} but must be 0 when sgdByteLength is 0."
```
```C++
// Current
"Invalid KTXwriter metadata. The value is missing the NULL terminator.",
"The last byte of the value must be a NULL terminator."

// Proposed 1
"Invalid KTXwriter metadata.",
"The last byte of the value must be a NULL terminator."

// Proposed 2
"KTXwriter metadata is missing the NULL terminator.",
"The last byte of the value must be a NULL terminator."

// Proposed 3
"KTXwriter missing NULL",
"The last byte of the value must be a NULL terminator."
```

However in some cases this could lower clarity or make multiple message (nearly) identical (while they are currently clear and unique).

Getting this format right is tricky, but I welcome any feedback or discussion on this topic to find the right one.

Reply:
I like Proposed 2 in both cases. The nearly-identical-messages issue can be avoided if the static message has a good  level of detail such as you have in Proposed 2.

I suggest you rework the messages along these lines then we can review them all, maybe in a conference call.

Reply:
I will do a rework pass, and when it is done we can have a conference call.

Reply:
We will delay the error message rework pass and talk about this topic more in our upcoming meeting.
These can be changed later with ease and they are a non-blocking matter.